pkg/scale/decode.go

```diff
@@ -108,6 +109,20 @@ type decodeState struct {
 }
 
 func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
+	// Handle BTreeMap type separately for the following reasons:
+	// 1. BTreeMap is a generic type, so we can't use the normal type switch
+	// 2. We cannot use BTreeCodec because we are comparing the type of the dstv.Interface() in the type switch
+	if isBTree(dstv.Type()) {
```

Comment:
Nit/Perhaps can rename to something the describes that it checks for both `btree.Btree` as well as `btree.Map`?

Reply:
hmm, both are based on the fundamental B-tree structure. I feel using `isBTree()` encapsulates the idea that it is checking for a type that is based on the B-tree structure, whether it's the tree itself or a map that uses it.

Reply:
Makes sense to me!

Reply:
I'll update the comments to capture this. Thank you!